oauth-common/src/main/java/io/strimzi/kafka/oauth/common/JSONUtil.java

```diff
@@ -87,18 +89,23 @@ public static String getClaimFromJWT(JsonNode node, String... path) {
         return node.asText();
     }
 
-    public static List<String> asListOfString(JsonNode arrayNode) {
+    public static List<String> asListOfString(JsonNode arrayOrString) {
+        return asListOfString(arrayOrString, " ");
+    }
+
+    public static List<String> asListOfString(JsonNode arrayOrString, String delimiter) {
 
         ArrayList<String> result = new ArrayList<>();
 
-        if (arrayNode.isTextual()) {
-            result.addAll(Arrays.asList(arrayNode.asText().split(" ")));
+        if (arrayOrString.isTextual()) {
+            result.addAll(Arrays.asList(arrayOrString.asText().split(Pattern.quote(delimiter)))
+                    .stream().map(String::trim).collect(Collectors.toList()));
```

Comment:
Do we want to filter out empty strings (which would happen if you had two delimiters next to each other)?

Reply:
This method is a general purpose method, so we probably don't want to remove empty strings there. The groups extraction usage makes sure to ignore empty strings.  
See:
- https://github.com/strimzi/strimzi-kafka-oauth/pull/133/files#diff-1af00685f38f1cd3d7e842281c5388d45ca757226cf5cf87bf81c41c597f1a0eR396
- https://github.com/strimzi/strimzi-kafka-oauth/pull/133/files#diff-c6b310d93d0ab3ac6eae52a71596528d2a1fc2f31b6def041735574ca443305bR282

Reply:
OK, but if it's really supposed to be general purpose you should define the contract in some javadoc. The possibility of delims next to each other is easily overlooked (in the common case where you're expecting a list of non-empty tokens).

Reply:
Added JavaDoc to JSONUtil.java